### DISCUSSION TITLE
Invalid us-emf schema when using validator (staging instance) and Hale studio

### DISCUSSION THREAD

**Discussion by vaneckch** [INSPIRE-MIF/helpdesk#106] [Q&A] [answered]
Good afternoon,

I have been attempting to create an INSPIRE conform dataset GML for Environmental Management Facilities (theme Utility and Governmental Services). 

The INSPIRE conform dataset is being created with the software HALE Studio. 
Here the first problem arises, the target schema INSPIRE preset available for us-emf seems to be an invalid target schema. It gives the following error in HALE studio:
Element definition without an associated type: {http://inspire.ec.europa.eu/schemas/us-emf/4.0)parentFacility.
This has been solved by using a draft schema provided with the following url:
https://inspire.ec.europa.eu/draft-schemas/us-emf/3.0rc3/EnvironmentalManagementFacilities.xsd

When validating the output (I seem to be only able to run the validation through the staging instance http://staging-inspire-validator.eu-west-1.elasticbeanstalk.com/etf-webapp/test-selection/index.html), the GML encounters the error that it is not using an official INSPIRE target schema and suggests using the one provided by the URL: 
https://inspire.ec.europa.eu/schemas/us-emf/4.0/EnvironmentalManagementFacilities.xsd

Complete report: http://staging-inspire-validator.eu-west-1.elasticbeanstalk.com/etf-webapp/test-run/index.html?id=EID72e06e44-d515-4e04-aa0a-57c7e59c31d9

Only this official target schema seems to be the same as I tried to import through HALE studio and gives the “Element definition without an associated type: {http://inspire.ec.europa.eu/schemas/us-emf/4.0)parentFacility.” issue by import. 

Is there something I am overlooking? Or is there not yet a functioning target schema available for US-emf? 

I hope someone could give me more insight in this issue. Thanks in advance.

Kind regards, 
Christel van Eck

Attached:  zipped GML
[us_stortplaatsen_us_draft_schema.zip](https://github.com/INSPIRE-MIF/helpdesk/files/9000569/us_stortplaatsen_us_draft_schema.zip)

PS. I am aware there are still some issues with the validity of the geometry and the cardinality of the attribute “function”.

**Comment by sMorrone** [comment, marked as answer]
Dear @vaneckch 
I just tried uploading the schema in hale studio v4.1 ( using upload via URL and schema location https://inspire.ec.europa.eu/schemas/us-emf/4.0/EnvironmentalManagementFacilities.xsd)  and it works correctly (see screenshot below)
![image](https://user-images.githubusercontent.com/13329248/176277721-d29a83c1-13f6-4976-be72-ab1af7ee6180.png)
I think there is an issue with hale studio presets not being up to date. The issue should already have been reported to hale developers (@thorsten-reitz could you please check? thx). However, there's a simple workaround i.e., to upload the desired schemas directly from the official INSPIRE repository https://inspire.ec.europa.eu/schemas/.
![image](https://user-images.githubusercontent.com/13329248/176280401-c6e22a4f-0fa1-408e-a90a-2e39ca647d27.png)

Kind regards
Stefania

**Reply by vaneckch** [reply]
Dear @sMorrone,

I used a saved copy of https://inspire.ec.europa.eu/schemas/us-emf/4.0/EnvironmentalManagementFacilities.xsd and did an upload by "From file". I do not obtain errors anymore on the target schema when validating via https://inspire.ec.europa.eu/validator/home/index.html

Thank you all so much for the support.

Cheers,
Christel

**Comment by JohannaOtt** [comment]
Hi Christel, hi Stefania,

Just a short note: Using the URLs from the schema repository in hale studio can also lead to the cached schemas being used - see [this issue](https://github.com/halestudio/hale/issues/960) for more input and workarounds.


All the best

Johanna

**Reply by vaneckch** [reply]
Dear Johanna (@JohannaOtt ) and Stefania (@sMorrone ),

Thank you for the fast response.

I have upgraded HALE Studio to version 4.1 (I worked before on version 4.0), but the problem stays persistent, after importing by URL the schema https://inspire.ec.europa.eu/schemas/us-emf/4.0/EnvironmentalManagementFacilities.xsd. The schema seems to load as in the example by @sMorrone , but I get the following error in the log:
 
![afbeelding](https://user-images.githubusercontent.com/44265062/176389081-fabe314a-a15a-43cf-ba35-41c98d28d75d.png)

When I remove the eu.esdihumboldt.util.resource.schemas.inspire_2021.3.9.jar as suggested as a solution in https://github.com/halestudio/hale/issues/960 I can’t seem to load the https://inspire.ec.europa.eu/schemas/us-emf/4.0/EnvironmentalManagementFacilities.xsd at all:

![afbeelding](https://user-images.githubusercontent.com/44265062/176389255-9d991d3d-5fdc-4ad4-be44-e131c7011571.png)

Is there still something going wrong on my side, or is the solution to wait for an update of the presets in HALE studio (as I understand that is a pending task?).

Kind regards,
Christel

**Reply by fabiovinci** [reply]
Dear @vaneckch,

as indicated by @JohannaOtt, also importing the schema from URL, hale uses the cached version.

I tried to remove the "eu.esdihumboldt.util.resource.schemas.inspire_2021.3.9.jar" file and I was able to load the latest schema version from the URL without errors.

If it does not work for you, a workaround is to save the schema as a file and import it through the "From file" option.

**Reply by vaneckch** [reply]
Dear @fabiovinci ,

Thank you for your response to the thread. Your workaround allowed me to create a valid INSPIRE GML in terms of the target schema. 

I'm not sure how to proceed with accepting the right answer as I can't seem to select a reply within a thread. So I'll accept the first answer by @sMorrone and add a reply to say that I used the downloaded .xsd and imported it with the 'From file' option.

Thank you all for the help.

Kind regards,
Christel

**Comment by sMorrone** [comment]
Thanks Johanna!
any news about the possibility that Hale Studio will keep up with the latest schema versions ? 
According to [the governance and release process document for INSPIRE schemas](https://github.com/INSPIRE-MIF/application-schemas/blob/main/governance-release-process/process.md) the release of the endorsed changes to the INSPIRE schemas happens twice a year (scheduled to take place by January 31 and July 31) and  "new bugfix and minor versions **will not trigger a change in the schema location URL**, meaning that particular attention shall be paid to breaking changes that might impact existing implementations". In the case reported by @vaneckch,  issue with hale preset is that the same URL (i.e. [https://inspire.ec.europa.eu/schemas/us-emf/4.0/EnvironmentalManagementFacilities.xsd ](https://inspire.ec.europa.eu/schemas/us-emf/4.0/EnvironmentalManagementFacilities.xsd)) now returns an updated schema version (see [release 2021.2 changelog](https://github.com/INSPIRE-MIF/application-schemas/releases/tag/2021.2))

**Reply by JohannaOtt** [reply]
Hi Stefania,
There was an internal ticket to do that but I created a [public issue](https://github.com/halestudio/hale/issues/977) on this now so that everyone can follow the development there (or do it themselves).

All the best
Johanna
